sdlib/d/gc/tcache.d

```diff
@@ -118,8 +119,9 @@ public:
 		if (pd.isSlab()) {
 			auto newSizeClass = getSizeClass(size);
 			auto oldSizeClass = pd.sizeClass;
-			if (samePointerness && newSizeClass == oldSizeClass) {
-				setSmallUsedCapacity(pd, ptr, size);
+			auto si = SlabAllocInfo(pd, ptr);
+			if ((samePointerness && newSizeClass == oldSizeClass)
+				    && (!si.allowsMetaData || si.setUsedCapacity(size))) {
```

Comment:
It is the right thing to do, but it is a change of behavior, and there is no change in the tests, meaning that behavior was untested. There needs to be a test for this.